lib/cloudregister/registerutils.py

```diff
@@ -173,6 +175,110 @@ def clear_rmt_as_scc_proxy_flag():
         pass
 
 
+# ----------------------------------------------------------------------------
+def clean_non_free_extensions():
+    """Remove non free extensions files from /etc/products.d/."""
+    extensions = get_extensions()
+    installed_products = get_installed_products()
+
+    for extension in extensions:
+        # The following not free condition checks on
+        # the access capability of the extension
+        # This is a server side capability
+        if not extension.get('free', True):
+            triplet = get_product_triplet(extension)
+            triplet = '{name}/{version}/{arch}'.format(
+                name=triplet.name,
+                version=triplet.version,
+                arch=triplet.arch
+            )
+            if triplet in installed_products:
+                return_code = exec_subprocess(
+                    ['SUSEConnect', '-d', '-p', triplet]
+                )
```

Comment:
For calling SUSEConnect I wrote a wrapper method named `run_SUSEConnect`. I believe it would be good if any SUSEConnect call is handled by the wrapper. One reason to do so is because there is a difference on `SUSEConnect` vs `transactional register`. I guess there is also a `transactional deregister`

Reply:
I thought to use it but refrained myself to move another method to `registerutils.py`

Reply:
will adapt the code (move the method implementation and more changes) to use it in `registerutils`

Reply:
Done in [f2e9acc](https://github.com/SUSE-Enceladus/cloud-regionsrv-client/pull/208/commits/f2e9acc78ba8cdcced60e2dd0cd9281707abac17) following the doc:

```bash
# De-register the system and base product
transactional-update register -d
 
# De-register a single product/extension
transactional-update register -d -p SUSE-Manager-Server/5.0/x86_64
```
Failed because of the linter

Reply:
nice :+1: